community/ndp/runtime/src/main/java/org/neo4j/ndp/runtime/internal/engine/QueryEngine.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ndp.runtime.internal.engine;
+
+import java.util.Map;
+
+import org.neo4j.kernel.api.exceptions.KernelException;
+import org.neo4j.stream.RecordStream;
+import org.neo4j.ndp.runtime.internal.session.SessionState;
+
+/**
+ * A runtime handler can handle a textual input language, yielding results. Query engines are not expected to be
+ * thread safe, each worker thread will have one query engine instance.
+ */
+public interface QueryEngine
```

Comment:
Ya, this is another component, like `Runtime` where we're still looking to find a good name. Agree QueryEngine is not helpful. I'll revert it back to StatementRunner for now.